pkg/kubelet/cadvisor/cadvisor_linux.go

```diff
@@ -70,7 +71,27 @@ func init() {
 	}
 }
 
-// Creates a cAdvisor and exports its API on the specified port if port > 0.
+func containerLabels(c *cadvisorapi.ContainerInfo) map[string]string {
+	set := map[string]string{"id": c.Name}
+	if len(c.Aliases) > 0 {
+		set["name"] = c.Aliases[0]
+	}
+	if image := c.Spec.Image; len(image) > 0 {
+		set["image"] = image
+	}
+	if v, ok := c.Spec.Labels[types.KubernetesPodNameLabel]; ok {
+		set["pod_name"] = v
+	}
```

Comment:
I've mentioned this in the PR description. Every additional label makes it a bit harder to work with the metrics. As id and pod_name are already unique, I didn't see the benefit of adding yet another label for the same purpose. Additionally, the pod UID is also part of the name label.

More explanation on [Target Labels should be Minimal](http://www.robustperception.io/target-labels-are-for-life-not-just-for-christmas/).